modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/components/VisibleSelectInput.es.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import classNames from 'classnames';
+import React, {forwardRef} from 'react';
+
+const LabelOptionListItem = ({onCloseButtonClicked, option}) => (
+	<li>
+		<ClayLabel
+			className="ddm-select-option-label text-truncate"
+			closeButtonProps={{
+				onClick: event => {
+					event.preventDefault();
+					event.stopPropagation();
+
+					onCloseButtonClicked({event, value: option.value});
+				},
+			}}
+			value={option.value}
+		>
+			{option.label}
+		</ClayLabel>
+	</li>
+);
+
+const OptionSelected = ({isPlaceholder, label}) => (
+	<div
+		className={classNames('option-selected', {
+			'option-selected-placeholder': isPlaceholder,
+		})}
+	>
+		{label}
+	</div>
+);
+
+const VisibleSelectInput = forwardRef(
+	(
+		{
+			className,
+			id,
+			multiple,
+			onClick,
+			onCloseButtonClicked,
+			options,
+			readOnly,
+			value,
+		},
+		ref
+	) => {
+		const triggerPlaceholder = multiple
+			? Liferay.Language.get('choose-options')
+			: Liferay.Language.get('choose-an-option');
+
+		return (
+			<div
+				className={classNames(
+					className,
+					'form-builder-select-field input-group-container'
+				)}
+				onClick={onClick}
+				ref={ref}
+			>
+				<ul
+					className={classNames(
+						'form-control results-chosen select-field-trigger',
+						{
+							'multiple-label-list': multiple,
+						}
+					)}
+					disabled={readOnly}
+					id={id}
+				>
```

Comment:
In summary of the comments above I think this could look like this:

```jsx
{(value.length === 0 || value.length === 1 && !multiple) ? (
	<OptionSelected
		isPlaceholder={value.length === 0 ? true : false}
		// This can go to a variable
		label={value.length === 0 ? triggerPlaceholder : options.find(option => option.value === value[0]).label}
	/>
) : value.map(item => {
	const option = options.find(option => option.value === item);
	return (
		<LabelOptionListItem
			key={`${option.value}-${option.label}`}
			onCloseButtonClicked={
				onCloseButtonClicked
			}
			option={option}
		/>
	);
})}
```